components/core/src/clp/clo/CommandLineArguments.cpp

```diff
@@ -273,6 +298,20 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
         if (m_max_num_results == 0) {
             throw invalid_argument("Max number of results cannot be 0.");
         }
+
+        if (m_count) {
+            if (m_reducer_host.empty()) {
```

Comment:
* We can use `parsed_command_line_options.count("reducer-host")` to validate if the user specified an arg rather than checking against the default value.
* We should print different errors if the user entered invalid option.
* We should validate that the output controls are mutually exclusive with the aggregation options.
  * On that note, there is an argument to be made that the reducer address and job ID are part of the output controls and really only the count option is an aggregation option.